tools/pythonpkg/duckdb_python.cpp

```diff
@@ -68,6 +72,15 @@ static void InitializeConnectionMethods(py::module_ &m) {
 	      py::arg("connection") = py::none())
 	    .def("duplicate", &PyConnectionWrapper::Cursor, "Create a duplicate of the current connection",
 	         py::arg("connection") = py::none());
+	m.def("register_scalar_udf", &PyConnectionWrapper::RegisterScalarUDF,
```

Comment:
`Scalar UDF` is database nomenclature that many users might not be familiar with - the SQLite API uses `create_function`. Perhaps `register_function` and `unregister_function` are better names here?

Reply:
If sqlite uses `create_function` we could use the same and `remove_function` no?
Although register is more duckdb-ish

Reply:
I'm fine with that as well - the main argument against is that our function signature is slightly different (SQLite only requires the number of arguments and no types) - but I think we want users to specify types

Reply:
The main reason I went with the `_udf` postfix is that we will likely want to add a table function variant later, and we can't infer if the passed in function is scalar/table.
I'd prefer to not add a table/scalar parameter to this function, as it doesn't make too much sense to share the same prototype

So `register_scalar` would be weird because we can't use `register_table`, that would become `register_table_[function|udf]`

Reply:
`UDF` could also refer to table functions - so at a minimum I would change it to `create_scalar_function` over `register_scalar_udf`.

I think having `create_function` refer to scalar functions is fine - as that is mostly the standard in all systems. Table functions and aggregate functions are generally harder to define as they have more options on how they can be executed.